src/clients/jobManagerWrapper.ts

```diff
@@ -169,6 +169,29 @@ export class JobManagerWrapper extends JobManagerClient {
     });
   }
 
+  public async validateAndUpdateExpiration(jobId: string): Promise<void> {
+    const getJobUrl = `/jobs/${jobId}`;
+    const newExpirationDate = new Date();
+    newExpirationDate.setDate(newExpirationDate.getDate() + this.expirationDays);
+
+    const job = await this.get<JobResponse | undefined>(getJobUrl);
+    if (job !== undefined) {
+      const expirationDate = new Date(job.expirationDate as Date);
+      if (expirationDate < newExpirationDate) {
+        await this.updateJobExpirationDate(jobId);
```

Comment:
pass newExpirationDate as a parameter to the updateJobExpirationDate method
or replace it with
await this.put(getJobUrl, {
      newExpirationDate,
    });
   // where - getJobUrl needs to be renamed to getOrUpdateURL